user_tools/src/spark_rapids_pytools/ascli_cli/ascli.py

```diff
@@ -0,0 +1,174 @@
+                are "Not Applicable"
+        """
+        runtime_platform = CloudPlatform.fromstring(platform)
+        if runtime_platform == CloudPlatform.ONPREM:
+            # if target_platform is specified, check if it's valid supported platform and filter the
+            # apps based on savings
+            if target_platform is not None:
+                if CliOnpremLocalMode.is_target_platform_supported(target_platform):
+                    if cluster is None:
+                        raise RuntimeError('OnPrem\'s cluster property file required to calculate'
+                                           'savings for ' + target_platform + ' platform')
+                else:
+                    raise RuntimeError(f'The platform [{target_platform}] is currently not supported'
+                                       'to calculate savings from OnPrem cluster')
+            else:
+                # For onPRem runtime, the filter should be reset to speedups when no target_platform
+                # is defined
+                filter_apps = QualFilterApp.tostring(QualFilterApp.SPEEDUPS)
+
+        wrapper_qual_options = {
+            'platformOpts': {
+                'credentialFile': None,
+                'deployMode': DeployMode.LOCAL,
+                'targetPlatform': target_platform
+            },
+            'migrationClustersProps': {
+                'cpuCluster': cluster,
+                'gpuCluster': None
+            },
+            'jobSubmissionProps': {
+                'remoteFolder': None,
+                'platformArgs': {
+                    'jvmMaxHeapSize': 24
+                }
+            },
+            'eventlogs': eventlogs,
+            'filterApps': filter_apps,
+            'toolsJar': None,
+            'gpuClusterRecommendation': gpu_cluster_recommendation,
+            'target_platform': target_platform
+        }
+        tool_obj = QualificationAsLocal(platform_type=runtime_platform,
+                                        output_folder=output_folder,
+                                        wrapper_options=wrapper_qual_options)
+        return tool_obj.launch()
+
+    def profiling(self,
+                  eventlogs: str = None,
+                  cluster: str = None,
+                  platform: str = CloudPlatform.tostring(CloudPlatform.get_default()),
+                  output_folder: str = None):
+        """Provides a wrapper to simplify the execution of RAPIDS Profiling tool.
+
+        The Profiling tool analyzes GPU event logs and generates information
+        which can be used for debugging and profiling Apache Spark applications.
+
+        :param eventlogs: Event log filenames or CSP storage directories containing event logs
+                (comma separated).
+
+                Skipping this argument requires that the cluster argument points to a valid
+                cluster name on the CSP.
+        :param cluster: The cluster on which the Apache Spark applications were executed.
```

Comment:
For Onprem platform, this becomes optional as the plan is to provide cluster configuration through worker_info.yaml file. PR to add user-tool profiling command for onprem is -https://github.com/NVIDIA/spark-rapids-tools/pull/431/files.
I don't see a point of adding cluster argument for Onprem platform. Please  take a look at the  PR and we can discuss if cluster is required or should we make it optional and change here to include worker_info.yaml.

Reply:
Thanks @nartal1 
You are right. The target is to combine those two parameters into a single one.
In an incremental improvement, the wrapper would be recognize the different formats and eventually consume the path as a worker-info file.